libraries/AP_Common/AP_Common.h

```diff
@@ -136,3 +136,18 @@ bool is_bounded_int32(int32_t value, int32_t lower_bound, int32_t upper_bound);
 #else
 #define SITL_printf(fmt, args ...)
 #endif
+
+
+/*
+ * Singleton macros
+ */
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+  #define SINGLETON_INIT_FAIL_MSG()     AP_HAL::panic("%s must be singleton", __FUNCTION__);
```

Comment:
Example of what this looks like on error?

'though... who really cares?  It's a "shouldn't happen" anyway.

Reply:
I'd like this to be panic on all boards